13-i-can-has-extension.md

```diff
@@ -0,0 +1,191 @@
+by individual implementations (and forks thereof),
+without requiring a centralized location to allocate feature bits or
+similar common resources.
+
+The intent is to allow node software to develop their own extensions
+to the protocol,
+to be later proposed to the BOLT spec once experience in live use
+has ironed out any design problems.
+
+# Table of Contents
+
+ * [Extension Encoding](#extension-encoding)
+ * [Message Type Reservation](#message-type-reservation)
+ * [Informing Extensions](#informing-extensions)
+   * [The `i_has_a_extensions` Message](#the-i_has_a_extensions-message)
+ * [Enabling Extensions](#enabling-extensions)
+   * [The `i_can_has_extension` Message](#the-i_can_has_extension-message)
+ * [Authors](#authors)
+
+# Extension Encoding
+
+Experimental features
+("extensions" in the terminology of this document)
+are identified by an extension code, a 20-byte vector of a
+white noise random distribution.
+
+An extension code SHOULD be selected
+by using RIPEMD160 on a vector generated by the concatenation of:
+
+* A human-readable UTF-8 description of the experimental feature.
+* A version number for the feature protocol.
+* A 20-byte salt from a white noise random distribution.
+
+By use of the above, developers of experimental features can
+acquire a randomly-selected extension code
+without permission from any central authority or group.
+
+An extension code MAY be selected by simply sampling twenty bytes
+from a white noise random distribution.
+
+The extension code composed of 20 `0x00` bytes is reserved to mean
+"no extension".
+
+# Message Type Reservation
+
+Message types 49152 (`0xC000`) to 65535 (`0xFFFF`) are reserved
+by this BOLT specification for extension messages.
+Two message types, `i_can_has_extension` (type 49152) and
+`i_has_a_extensions` (type 49153) are used to inform about extensions
+and enable an extension to be used over the peer protocol.
+
+Message types 49154 (`0xC002`) to 65535 (`0xFFFF`) are to be used
+for extension messages.
+These message types are _multiplexed_,
+meaning that all extensions share the same message type numbers.
+Messages for a specific extension are called _extension-specific_
+messages.
+
+When a node receives a message of type 49154 to 65535,
+that message will have a meaning and format
+according to the "currently enabled extension" of the receiver.
+
+If the "currently enabled extension" is "no extension",
+then message types 49154 to 65535 will result in undefined behavior.
+
+Each node has its own concept of "currently enabled extension",
+which it uses for received messages.
+The `i_can_has_extension` message is used by the sender
+to change the "currently enabled extension" of the receiver.
+
+## Requirements
+
+Before sending an extension-specific message to the peer,
+the sending node MUST ensure that
+the "currently enabled extension" of the receiver is the extension
+the sending node intends to use.
```

Comment:
as mentioned above how can that go without handshake and a "connection"?

Reply:
BOLT 8 already specifies an ongoing connection; all Lightning messages are sent via BOLT 8.